docs/src/user_docs/stdlib/mem.md

```diff
@@ -4,3 +4,6 @@ Module `std::mem` contains a set of utility procedures for working with random a
 | Procedure | Description |
 | ----------- | ------------- |
 | memcopy | Copies `n` words from `read_ptr` to `write_ptr`.<br /><br />Stack transition looks as follows:<br /><br />[n, read_ptr, write_ptr, ...] -> [...]<br /><br />Cycles: 15 + 16n |
+| pipe_double_words_to_memory | Copies an even number of words from the advice_stack to memory.<br /><br />Input: [C, B, A, write_ptr, end_ptr, ...]<br />Output: [C, B, A, write_ptr, ...]<br /><br />Where:<br />- The words C, B, and A are the RPO hasher state<br />- A is the capacity<br />- C, B are the rate portion of the state<br />- The value `words = end_ptr - write_ptr` must be positive and even<br /><br />Cycles: 10 + 9 * word_pairs |
+| pipe_words_to_memory | Copies an arbitrary number of words from the advice stack to memory<br /><br />Input: [num_words, write_ptr, ...]<br />Output: [HASH, write_ptr', ...]<br /><br />Cycles:<br />even num_words: 48 + 9 * num_words / 2<br />odd num_words: 65 + 9 * round_down(num_words / 2) |
```

Comment:
nit: I think this is a bit confusing, since `HASH` in the output state isn't explained anywhere.